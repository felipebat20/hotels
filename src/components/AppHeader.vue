<template>
  <div class="row items-center q-mx-auto q-pa-xs z-max">
    <h1>
      <img :src="Logo"
        alt="Logo"
        :style="{ width: '200px' }"
        class="q-ma-none"
      />
    </h1>
  </div>
</template>

<script setup lang="ts">
import Logo from '@/assets/main-logo.svg'
</script>
