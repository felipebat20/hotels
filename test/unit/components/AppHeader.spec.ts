import { installQuasarPlugin } from '@quasar/quasar-app-extension-testing-unit-vitest';
import { mount } from '@vue/test-utils';
import { describe, expect, it, vi } from 'vitest';
import { createTestingPinia } from '@pinia/testing'

import AppHeader from '@/components/AppHeader.vue';

installQuasarPlugin();

describe('test AppHeader', () => {
  const createWrapper = () => {
    return mount(AppHeader, {
      global: {
        plugins: [createTestingPinia({ createSpy: vi.fn })],
      },
    });
  };

  it('should mount component properly', () => {
    const wrapper = createWrapper();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
