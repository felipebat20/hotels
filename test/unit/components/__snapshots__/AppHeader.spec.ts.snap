// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test AppHeader > should mount component properly 1`] = `
"<div class="row items-center q-mx-auto q-pa-xs z-max">
  <h1><img src="/src/assets/main-logo.svg" alt="Logo" style="width: 200px;" class="q-ma-none"></h1>
</div>"
`;
